### DISCUSSION TITLE
Patched font has different settings?

### DISCUSSION THREAD

**Discussion by rieje** [ryanoasis/nerd-fonts#805] [Q&A]
Shouldn't the patched version of a nerd font e.g. jetbrains-mono be exactly the same but with support for glyphs? I replaced jetbrains-mono with the patched version for my terminal (Alacritty) and as soon as I save the Alacritty config (the changes are applied live), something changed (looks like height because the line numbers changed, but with all the different font settings probably other things as well).

I want it to be the same as the developer intended, is this possible? Should the patched font in theory be a complete drop-in replacement for the original font or are there some caveats?

**Comment by Finii** [comment]
Ideally the patched font should look the same (i.e. be identical) to the source font, just contain the extra glyphs (symbols).

Well, sometimes there are exceptions, but that should be ... exceptional. Affect only some glyphs.

What I do not understand is the "different settings that the font has". I thought settings are stuff the user sets, but well...
If there is a visible difference please raise an Issue.
* Make sure you compare the same fonts (i.e. original jetbrains-mono changes over time)
* If possible state download links and exact file names of both fonts
* Add screenshots with the issue, preferable where we can see the terminal window itself and not just a small snippet
* Add (i.e. copy&paste) affected glyphs directly or state glyph codepoint numbers
* Show your terminal font setting as screenshot (well, iirc Alacritty has no font settings dialog, so the config in text is ok there)

A bug that can not be reproduced can most likely not be fixed.

Some more hints:
* Try the fonts from the Release Candidate, maybe the bug is fixed already
* The matching source font versions are in this repo (`src/unpatched-fonts/`)
* If it affects only very small font sizes, maybe it is #761

**Reply by Finii** [reply]
Due to some `fontforge` ... peculiarity we can not (with just `fontforge`) keep these settings in the font-file:
```
$ showttf -c src/unpatched-fonts/JetBrainsMono/Ligatures/Regular/JetBrainsMono-Regular.ttf | grep '(flags|ppem)='
	flags=f baseline_at_0 lsb_at_0 instrs_depend_on_size ppem_to_int 
	lowestppem=6
$ showttf -c patched-fonts/JetBrainsMono/Ligatures/Regular/complete/JetBrains\ Mono\ Regular\ Nerd\ Font\ Complete.ttf | grep '(flags|ppem)='
	flags=1f baseline_at_0 lsb_at_0 instrs_depend_on_size ppem_to_int instr_set_width 
	lowestppem=8
```

**Reply by rieje** [reply]
My problem was simply not including the suffix "Mono" in "JetBrainsMono Nerd Font"... the package I installed from the distro led me to believe all the provided fonts were monospace and the "Mono" "JetBrainsMono" too. Confirmed there's no apparent differences as expected.

Appreciate your answer. Two more noob questions:

* It seems only .ttf files are provided at least for JetBrains Mono but source provides .otf as well. Are there any differences between the two? I thought .otf is preferable. Would the results be the same between the two formats or is it simply a difference in implementation?

* (Feel free to skip, just curious): Is it possible to treat match icons and the patched font separately in string involving both icons and text? I want to make the icons bigger than the text.

I have a string involving icons and text and I want just the text to be JetBrains Mono (setting to the patched font results in the icons a little too small, I guess this is by nature of monospace). So I was thinking of installing the non-monospace version of just the icons and have that match the icons first before the patched/unpatched JetBrains Mono matches the rest (in pango format `Symbols Nerd Font 20, JetBrainsMono 14` the first font specified takes priority i believe).

On Arch Linux, when I try to install `ttf-nerd-fonts-symbols` it conflicts with `nerd-fonts-jetbrains-mono` (not sure why, IIUC the former is just symbols non-monospaced while the latter is the patched JetBrains font monospaced). So I install Symbols-2048-em Nerd Font Complete.ttf manually. The result is the text and the icons match the patched/unpatched JetBrains Mono that's specified as the lower priority. The problem isn't a big deal, just curious where I'm off on understanding.

**Reply by Finii** [reply]
1. What I know, `otf` has some more features that `ttf` lacks. I believe there are no features in `ttf` that can not be represented in `otf`.
For fonts that do not use any of the additional `otf` features it does not matter which you choose.
https://www.coreldraw.com/en/tips/otf-vs-ttf/

2. I do not understand the beginning, but for the last sentence: Symbols are usually bigger in the 'non-mono' Nerd Fonts, ... well naming ist hard here. The fonts that are called `SomeName Nerd Font` and not `SomeName Nerd Font Mono`.
In your example `JetBrainMono Nerd Font` has bigger icons than `JetBrainMono Nerd Font Mono`.
In the `Nerd Font Mono` fonts the icons are as big as the capital `M` (approx), while in the `Nerd Font` (non-mono) they are usually 1.5 times the `M`.

3. These Arch packages turn up here rather frequently. (Un)fortunately we can do nothing about them and you should contact their maintainer in case of problems.

**Reply by Finii** [reply]
Further for 1.:

For example fonts with swashes are only available/possible as `otf`. For some projects I use `Ciao Bella`, here is an example with swashes (these long tails that swirl back to the beginning of the word):

![image](https://user-images.githubusercontent.com/16012374/157929387-ced9edec-c351-4a0b-88da-20e9d1c2ce3d.png)

Of yourse you need a program that can make use of the extended feature set. On Windows I used _Publisher_, and since some years that I switched to Linux I use _Scribus_. Some OTF features seem to trickle down into 'normal' applications now (after years and years) it seems, so ... lets hope we can at some point in time use em in _Impress_ or _Powerpoint_.

**Reply by Finii** [reply]
![image](https://user-images.githubusercontent.com/16012374/157931432-daabde2e-604f-417c-8809-e7ad0ae3bb50.png)
